csharp/src/Apache.Arrow.Adbc/Extensions/ListArrayExtensions.cs

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using Apache.Arrow.Types;
+
+namespace Apache.Arrow.Adbc.Extensions
+{
+    public static class ListArrayExtensions
+    {
+        /// <summary>
+        /// Creates a <see cref="ListArray"/> from a list of <see cref="IArrowArray"/> data for the given datatype <see cref="IArrowArray"/>.
+        /// </summary>
+        /// <param name="arrayList">The list of data.</param>
+        /// <param name="dataType">The data type of the contained data.</param>
+        /// <returns>A <see cref="ListArray"/> of the data.</returns>
+        public static ListArray CreateNestedListArray(this IReadOnlyList<IArrowArray?> arrayList, IArrowType dataType)
```

Comment:
The summary seems underspecified for a public function. I can't tell from the text what it does or what I need to pass to it to make it work.

Is this functionality which should be in the core Arrow library?

Reply:
Yes. I believe this should likely be included in the Arrow library. Not sure of it proper placement in the namespace, yet. Let me work on naming and documentation.